earth_enterprise/src/fusion/gecombineterrain/combineterrainprofiler.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 the Open GEE Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COMBINETERRAINPROFILER_H
+#define COMBINETERRAINPROFILER_H
+
+#include <string>
+
+#include "quadtreepath.h"
+
+/*
+ * Singleton class for profiling terrain combine events. The programmer can use
+ * this class to mark the beginning and end of events, and the timing of those
+ * events will be written out to a log file. This class is intended for
+ * performance debugging.
+ */
+class CombineTerrainProfiler {
+  public:
+    static CombineTerrainProfiler * instance() { return _instance; }
```

Comment:
May be here check if '_instance' is null and create the instance and if it's not null, simply return the instance.
This way the client code would need to call only 'instance()' static method.

Reply:
I actually considered that method but prefer the way I did it. If you create the `_instance` object here in an if statement it creates a race condition - if two threads call it at the same time they might create two separate instances. Instead, I opted to make `_instance` a static const member variable, which ensures that it will be instantiated once when the program starts.

Reply:
you can create a thread safe singleton, but yeah the way you have will work too.